receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -362,43 +366,57 @@ func (jr *jReceiver) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest)
 }
 
 func (jr *jReceiver) startAgent(_ receiver.Host) error {
-	processorConfigs := []agentapp.ProcessorConfiguration{
-		{
-			// Compact Thrift running by default on 6831.
-			Model:    "jaeger",
-			Protocol: "compact",
-			Server: agentapp.ServerConfiguration{
-				HostPort: jr.agentCompactThriftAddr(),
-			},
-		},
-		{
-			// Binary Thrift running by default on 6832.
-			Model:    "jaeger",
-			Protocol: "binary",
-			Server: agentapp.ServerConfiguration{
-				HostPort: jr.agentBinaryThriftAddr(),
-			},
-		},
+	if !jr.agentBinaryThriftEnabled() && !jr.agentCompactThriftEnabled() && !jr.agentHTTPEnabled() {
+		return nil
 	}
 
-	builder := agentapp.Builder{
-		Processors: processorConfigs,
-		HTTPServer: agentapp.HTTPServerConfiguration{
-			HostPort: jr.agentAddress(),
-		},
+	handler := jaegerThrift.NewAgentProcessor(jr)
+
+	if jr.agentBinaryThriftEnabled() {
+		transport, err := thriftudp.NewTUDPServerTransport(jr.agentBinaryThriftAddr())
+		if err != nil {
+			return err
+		}
+		server, err := servers.NewTBufferedServer(transport, defaultAgentQueueSize, defaultAgentMaxPacketSize, metrics.NullFactory)
+		if err != nil {
+			return err
+		}
+		processor, err := processors.NewThriftProcessor(server, defaultAgentServerWorkers, metrics.NullFactory, apacheThrift.NewTBinaryProtocolFactoryDefault(), handler, jr.logger)
+		if err != nil {
+			return err
+		}
+		jr.agentProcessors = append(jr.agentProcessors, processor)
 	}
 
-	agent, err := builder.CreateAgent(jr, zap.NewNop(), metrics.NullFactory)
-	if err != nil {
-		return err
+	if jr.agentCompactThriftEnabled() {
+		transport, err := thriftudp.NewTUDPServerTransport(jr.agentCompactThriftAddr())
```

Comment:
Except for the simmetrical calls to `jr.agentCompactThriftAddr()` vs `jr.agentBinaryThriftAddr()` and `apacheThrift.NewTCompactProtocolFactory()` vs `apacheThrift.NewTBinaryProtocolFactoryDefault()`, this block is equal to the one above, extracting a function and passing the functions above seems reasonable.

Reply:
Thx for the suggestion.  Fixed.